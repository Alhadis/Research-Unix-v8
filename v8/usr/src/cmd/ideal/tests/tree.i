tree
.nf
.IS
box treenode
{
var ne, nw, sw, se,
    hook, lchild, rchild, cen;
ne = nw + width;
se = sw + width;
ne = se + (0,1)*height;
hook = (ne + nw)/2;
cen = (ne + sw)/2;
lchild = (sw + cen)/2;
rchild = (se + cen)/2;
conn ne to nw;
conn nw to sw;
conn sw to se;
conn se to ne;
conn (nw + sw)/2 to (ne + se)/2;
conn cen to (sw + se)/2;
}

box tree
{
var height, width;
height = ht/2;
width = 2*wd/3;
put node0:treenode{hook=root;};
put node1:treenode{hook=node0.hook-(0,1)*ht-wd;};
put node2:treenode{hook=node0.hook-(0,1)*ht+wd;};
put node3:treenode{hook=node2.hook-(0,1)*ht-wd;};
put node4:treenode{hook=node2.hook-(0,1)*ht+wd;};
conn node0.lchild to node1.hook;
conn node0.rchild to node2.hook;
conn node2.lchild to node3.hook;
conn node2.rchild to node4.hook;
}

box main
{
put T:tree{var root, ht, wd;
root = (1,-3);
ht = 1;
wd = 1;
};
}
.IE
.IS
...forget treenode tree
.IE
