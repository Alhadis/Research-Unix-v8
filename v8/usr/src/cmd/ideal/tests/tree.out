tree
.nf
.IS
...maxx 3.33333
...maxy -3
...minx -0.333333
...miny -5.5
...line 3 -5 2.16667 -4.375
...line 1 -5 1.83333 -4.375
...line 2 -4 1.16667 -3.375
...line 0 -4 0.833333 -3.375
...line 3 -5.5 3 -5.25
...line 3.33333 -5.25 2.66667 -5.25
...line 3.33333 -5 3.33333 -5.5
...line 3.33333 -5.5 2.66667 -5.5
...line 2.66667 -5.5 2.66667 -5
...line 2.66667 -5 3.33333 -5
...line 1 -5.5 1 -5.25
...line 1.33333 -5.25 0.666667 -5.25
...line 1.33333 -5 1.33333 -5.5
...line 1.33333 -5.5 0.666667 -5.5
...line 0.666667 -5.5 0.666667 -5
...line 0.666667 -5 1.33333 -5
...line 2 -4.5 2 -4.25
...line 2.33333 -4.25 1.66667 -4.25
...line 2.33333 -4 2.33333 -4.5
...line 2.33333 -4.5 1.66667 -4.5
...line 1.66667 -4.5 1.66667 -4
...line 1.66667 -4 2.33333 -4
...line 0 -4.5 0 -4.25
...line 0.333333 -4.25 -0.333333 -4.25
...line 0.333333 -4 0.333333 -4.5
...line 0.333333 -4.5 -0.333333 -4.5
...line -0.333333 -4.5 -0.333333 -4
...line -0.333333 -4 0.333333 -4
...line 1 -3.5 1 -3.25
...line 1.33333 -3.25 0.666667 -3.25
...line 1.33333 -3 1.33333 -3.5
...line 1.33333 -3.5 0.666667 -3.5
...line 0.666667 -3.5 0.666667 -3
...line 0.666667 -3 1.33333 -3
.IE
.IS
.IE
